use-shopping-cart/core/slice.js

```diff
@@ -106,6 +105,53 @@ const slice = createSlice({
       reducer: (state, { payload }) => {
         return removeEntry({ state, id: payload })
       }
+    },
+    loadCart: {
+      reducer: (state, { payload }) => {
+        const { cartDetails, shouldMerge } = payload
+        if (!shouldMerge) state = { ...initialState }
+
+        for (const id in cartDetails) {
+          const entry = cartDetails[id]
+          state = createEntry({
+            ...state,
+            state,
+            product: entry,
+            count: entry.quantity
+          })
+        }
+        return state
+      },
+      prepare: (cartDetails, shouldMerge = true) => {
+        return { payload: { cartDetails, shouldMerge } }
+      }
+    },
+    // TODO: discuss solutions to this. The two following actions do not redirect to checkout.
+    redirectToCheckout: {
+      reducer: (state) =>
+        checkoutHandler(state, {
+          modes: ['client-only', 'checkout-session'],
+          stripe(stripe, options) {
+            return stripe.redirectToCheckout(options)
+          }
+        })
+    },
+    checkoutSingleItem: {
+      reducer: (state) =>
+        checkoutHandler(state, {
+          modes: ['client-only'],
+          stripe(stripe, options, { sku, quantity = 1 }) {
+            options.lineItems = [{ price: sku, quantity }]
+            return stripe.redirectToCheckout(options)
+          }
+        })
+    },
+    handleCartClick: (state) => {
+      // TODO: Fix the following line. It is underlined by ESLint because it doesn't do anything.
+      !state.shouldDisplayCart
```

Comment:
I could use Immer's void, here to make it look like it's doing a thing, I think

Reply:
I'm not sure what Immer's void is. The issue here is just that you've made a function block and then just made a statement without any assignment or return value.

```js
(state) => {
  // Doesn't assign or return
  !state.shouldDisplayCart
}

(state) => {
  // Assignment works with Immer
  state.shouldDisplayCart = !state.shouldDisplayCart
  // or you could return a value. The Immer way is more succint though.
}
```

Reply:
Oh, I see... but wait, the tests I wrote for this still work 🤔  or maybe I need to re-examine that test. I'll double check, but I wouldn't have left that as is if the tests weren't passing. I'll take a look once this is merged

Reply:
@dayhaysoos Yeah, a good way to see what's going on, you can remove the code from your library/app and see if the tests fail to verify that the tests are in fact testing what they're supposed to. That or if you write the tests before the code, it's the same idea.